From a04b8e6dad13c897daea41a90115e5c39efbaab4 Mon Sep 17 00:00:00 2001
From: Zheng Chongzhen <zhengchongzhen@wxiat.com>
Date: Tue, 7 Jun 2022 17:41:58 +0800
Subject: [PATCH] sw64: fix compile error for CONFIG_PCI=n
Patch-mainline: NO, OTHERS
References: openEuler-22.03-LTS
Modified-by-SEL: No


Sunway inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I56QAM

--------------------------------

It fixes pci_domain_nr redefined error when CONFIG_PCI=n.

Signed-off-by: Zheng Chongzhen <zhengchongzhen@wxiat.com>

Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Acked-by: Xie XiuQi <xiexiuqi@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/Kconfig           | 3 +--
 arch/sw_64/include/asm/pci.h | 2 ++
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/arch/sw_64/Kconfig b/arch/sw_64/Kconfig
index 318a423d87dd..347958e60023 100644
--- a/arch/sw_64/Kconfig
+++ b/arch/sw_64/Kconfig
@@ -279,8 +279,7 @@ config PCI
 	  VESA. If you have PCI, say Y, otherwise N.
 
 config PCI_DOMAINS
-	bool
-	default y
+	def_bool PCI
 
 config PCI_SYSCALL
 	def_bool PCI
diff --git a/arch/sw_64/include/asm/pci.h b/arch/sw_64/include/asm/pci.h
index f1c291101811..fc6a4b469f30 100644
--- a/arch/sw_64/include/asm/pci.h
+++ b/arch/sw_64/include/asm/pci.h
@@ -77,6 +77,7 @@ extern struct pci_controller *hose_head;
 extern struct syscore_ops iommu_cpu_syscore_ops;
 #endif
 
+#ifdef CONFIG_PCI_DOMAINS
 static inline int pci_domain_nr(struct pci_bus *bus) { return 0; }
 
 static inline int pci_proc_domain(struct pci_bus *bus)
@@ -85,6 +86,7 @@ static inline int pci_proc_domain(struct pci_bus *bus)
 
 	return hose->need_domain_info;
 }
+#endif
 
 #ifdef CONFIG_NUMA
 static inline int __pcibus_to_node(const struct pci_bus *bus)
-- 
2.34.1

